### PR TITLE
Dev pvalidate

### PR DESCRIPTION
**Pull request by ziqiaozhou**

### CODE REVIEW

File: monitor-unverified/src/main.rs [hunk 1]
```diff
@@ -14,9 +14,34 @@ fn panic(_info: &PanicInfo) -> ! {
 
 global_asm!(include_str!("entry.s"));
```
Comment (wdcui) [inline monitor-unverified/src/main.rs:15]: Is entry.s a file? I don't see it in the PR.

File: monitor/src/main.rs [hunk 2]
```diff
@@ -1,100 +1,130 @@
 #![no_std] // don't link the Rust standard library
 #![no_main] // disable all Rust-level entry points
+#![feature(asm)]
 
 #[allow(unused_imports)]
 use builtin::*;
 #[allow(unused_imports)]
 use builtin_macros::*;
+#[allow(unused_imports)]
+mod pervasive;
```
Comment (wdcui) [inline monitor/src/main.rs:10]: the "pervasive" subfolder is removed in this PR?

File: monitor/src/main.rs [hunk 3]
```diff
@@ -1,100 +1,130 @@
...
     }
 }
 
 #[spec]
-pub fn mem_validated2(addr: u64) -> bool {
-    if addr < 0x40000 {
-        true
-    } else {
-        false
+pub fn state_to_rflags(addr: u64, state: Tracked<VMState>) -> u64{
```
Comment (wdcui) [inline monitor/src/main.rs:59]: what's this for?

File: monitor/src/main.rs (hunk 2 above)
Comment (ziqiaozhou) [reply monitor/src/main.rs:10]: It is linked to a folder in the module verus. builtin, builtin_macros, and pervasive defines useful function and macros used for verification purposes.

**Pull request merged**